src/source/JNIGenerator.scala

```diff
@@ -404,11 +406,11 @@ class JNIGenerator(spec: Spec) extends Generator(spec) {
 
   def toJniCall(ty: TypeRef, f: String => String): String = toJniCall(ty.resolved, f, false)
   def toJniCall(m: MExpr, f: String => String, needRef: Boolean): String = m.base match {
-    case p: MPrimitive => f(if (needRef) "Object" else IdentStyle.camelUpper(p.jName))
-    case MString => "(jstring)" + f("Object")
+    case p: MPrimitive => "(" + f(if (needRef) "Object" else IdentStyle.camelUpper(p.jName))
+    case MString => "reinterpret_cast<jstring>(" + f("Object")
```

Comment:
Generating the open-paren and close-paren at widely-separated places in the code seems confusing, and likely to miss some cases.  Why can't the close-paren be generated here so it's directly part of the cast expression?

Reply:
The problem lies in JNIGenerator.scala:288 where the expression to be casted is not entirely passed to the toJniCall method. I could add a function to generate the whole expression to be casted and then send that to toJniCall. It was not a problem before with c-style cast of course.

Reply:
So after looking at it again, it does not really make sense to try to do that. Line 288 generates the first line of the call: something like

`auto jret = reinterpret_cast<jstring>(jniEnv->CallObjectMethod(`

Then on line 294 it uses that string length to generate properly aligned method call so it means the output of toJniCall should depends on the length of its output... Hoping I am clear here ;)